### ISSUE TITLE
Clarify the term "current context" with regards to `sch:name` and `sch:value-of` inside an assertion message

### ISSUE THREAD

**Issue by dmj** [Schematron/schematron-enhancement-proposals#99]
Section 6.3 talks about the context of `sch:name` and `sch:value-of` inside an assertion message as follows:

> The content of the assertion message is reported as (svrl:failed-assert|svrl:successful-report)/ svrl:text. The context for evaluating elements name and value-of featuring as part of the message is the result of evaluating the first available attribute of the following sequence: a) the assertion's subject attribute; b) the containing rule's subject attribute; c) the containing rule's context attribute.

I.e. the context for evaluating `sch:name`/`sch:value-of` is defined to be the item selected by evaluating `@subject`, `../@subject` or `../@context`.

The definitions of `sch:name` and `sch:value-of` state the the respective expressions should be evaluated "in the current context".

What is the current context?

If we follow 6.3, then the current context for `sch:name` and `sch:value-of` is the result of evaluating the first of `@subject`, `../@subject` or `../@context`.

a) what is the context for evaluating `@subject`?
b) what is the context for evaluating `../@subject`?
c) do we really mean to evaluate `../@context` again? If so: what is the context for evaluating it? If not: Does it mean the node selected by the rule context expression -or- does it mean the items selected by evaluating `../@visit-each` (with a default of '.')?
d) maybe find better terms?

As a side node: Using the result of evaluating `@subject`, `../@subject` as context for `sch:name`, `sch:value-of` is backward-incompatible change, no?

**Comment by rjelliffe** [comment]
The @subject was intended to allow the ***@***.*** to identify
something more useful (in the view of the schema writer) than the context
taken from ***@***.*** The theory is that this gives the schema
writer more flexibility to organize their reporting more independently of
the rule structure.  (People will recall the problem that grammars have,
where a problem is reported as belonging to where the content model fails
not necessarily where the fault occurs.)

So the @name and @value operate off the context, not the subject.

For trivial example,
````
  <  sch:rule context="th"  >
     < sch:assert test="parent::table"  subject="parent::*"
            >A table header should only be in a table.< /sch:report>
 < /sch:rule>
````
It would indeed be possible/useful to have way to expose the subject
location to sch:name and sch:value: I don't recall what the mechanism
decided for visit-each was (built-in variable?)

It might be useful for the standard to say that the @subject should
identify a single node, and that e.g. @subjects that identify multiple
nodes result in error or undefined behaviour.

Rick
